qiita_db/artifact.py

```diff
@@ -512,21 +512,30 @@ def delete(cls, artifact_id):
                 raise qdb.exceptions.QiitaDBArtifactDeletionError(
                     artifact_id, "it is public")
 
-            # Check if this artifact has any children
-            if instance.children:
-                raise qdb.exceptions.QiitaDBArtifactDeletionError(
-                    artifact_id,
-                    "it has children: %s"
-                    % ', '.join([str(c.id) for c in instance.children]))
+            # delete children if children exists
+            for c in sorted(instance.children):
+                try:
+                    instance.delete(c.id)
+                except qdb.exceptions.QiitaDBArtifactDeletionError as e:
+                    msg = 'because children %s' % str(e)
```

Comment:
'because children %s' seems to potentially be lacking a verb, as far as error messages go

Reply:
...this might need to be clarified before approving.